src/payment/strategies/moneris/moneris-payment-strategy.ts

```diff
@@ -239,7 +239,11 @@ export default class MonerisPaymentStrategy implements PaymentStrategy {
         return iframe;
     }
 
-    private _handleMonerisResponse(response: MessageEvent): string {
+    private _handleMonerisResponse(response: MessageEvent): string | undefined {
+        if (typeof(response.data) === 'object') {
```

Comment:
What causes this malformed message?

Do we need to handle the scenario where we get an incorrect message, followed by a correct message?

Currently I think we will reject and fail after the first malformed message.

Reply:
Should we filter out malformed messages instead?

Currently it looks like a malformed message will lead to us setting `nonce` as `undefined`, not submitting payment and exiting execution.

Instead, should we not filter them out and only resolve/reject based on messages received in the correct format?

Reply:
@leeBigCommerce i moved the validation at the moment of the event listener so the promise will not be resolved until we get a proper response